projects/gloo/cli/pkg/cmd/gatewayapi/convert/write.go

```diff
@@ -0,0 +1,321 @@
+		if err != nil {
+			return err
+		}
+		if err := writeObjectToFile(opts, r, yml); err != nil {
+			return err
+		}
+	}
+	for _, r := range g.gatewayAPICache.Upstreams {
+		yml, err := yaml.Marshal(r.Upstream)
+		if err != nil {
+			return err
+		}
+		if err := writeObjectToFile(opts, r, yml); err != nil {
+			return err
+		}
+	}
+	for _, r := range g.gatewayAPICache.AuthConfigs {
+		yml, err := yaml.Marshal(r.AuthConfig)
+		if err != nil {
+			return err
+		}
+		if err := writeObjectToFile(opts, r, yml); err != nil {
+			return err
+		}
+	}
+	for _, r := range g.gatewayAPICache.ListenerSets {
+		yml, err := yaml.Marshal(r.XListenerSet)
+		if err != nil {
+			return err
+		}
+		if err := writeObjectToFile(opts, r, yml); err != nil {
+			return err
+		}
+	}
+	for _, r := range g.gatewayAPICache.Settings {
+		yml, err := yaml.Marshal(r.Settings)
+		if err != nil {
+			return err
+		}
+		if err := writeObjectToFile(opts, r, yml); err != nil {
+			return err
+		}
+	}
+	for _, r := range g.gatewayAPICache.YamlObjects {
+		yml, err := yaml.Marshal(r.Object)
+		if err != nil {
+			return err
+		}
+		if err := writeObjectToFile(opts, r, yml); err != nil {
+			return err
+		}
+	}
+	for _, r := range g.gatewayAPICache.DirectResponses {
+		yml, err := yaml.Marshal(r.DirectResponse)
+		if err != nil {
+			return err
+		}
+		if err := writeObjectToFile(opts, r, yml); err != nil {
+			return err
+		}
+	}
+	// create Gloo Errors directory and splt the files based on their error string.
+	folder, err := createSubDir(opts.OutputDir, "gloo-errors")
+	if err != nil {
+		return err
+	}
+	//organize all errors into a map
+
+	for t, errors := range g.errors {
+		f, err := os.Create(fmt.Sprintf("%s/%s.txt", folder, t))
+		if err != nil {
+			return err
+		}
+		errorsMap := make(map[string][]error)
+		for _, err := range errors {
+			// for each error organize by type and name string
+			key := fmt.Sprintf("[%s] %s/%s", err.crdType, err.namespace, err.name)
+			_, exist := errorsMap[key]
+			if !exist {
+				errorsMap[key] = make([]error, 0)
+			}
+			errorsMap[key] = append(errorsMap[key], err.err)
+		}
+		for key, errors := range errorsMap {
+			_, err := f.WriteString("\n" + key + "\n")
+			if err != nil {
+				return err
+			}
+			for _, gerr := range errors {
+				_, err := f.WriteString("\t" + gerr.Error() + "\n")
+				if err != nil {
+					return err
+				}
+			}
+		}
+
+		err = f.Close()
+		if err != nil {
+			return err
+		}
+	}
+	if len(g.errors) > 0 {
+		fmt.Printf("Errros were encountered during translation, please check %s/gloo-errors\n", opts.OutputDir)
+	}
+	fmt.Printf("Files succesfully written to %s\n", opts.OutputDir)
+
+	if opts.CreateNamespaces {
+		if err := g.createNamespaces(opts.OutputDir); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (g *GatewayAPIOutput) createNamespaces(dir string) error {
+	namespaces := map[string]bool{}
+	// iterate through all main objects and find unique namespaces, references shouldnt be needed
```

Comment:
Would it be better to get namespaces from all objects ? Would this break if a resource such as a non-http route exists in a separate namespace ?

Reply:
i didnt want to bloat this function with iterating through all objects because it would set the precedent that any new object needs to also be added here. This is more of a convenience feature for testing and if it becomes apparent that its missing a lot of namespaces then we can add more objects.